File: r/sorting_algorithm.r
bubble_sort <- function(arr) {
  n <- length(arr)
  repeat {
    swapped <- FALSE
    for (i in 1:(n-1)) {
      if (arr[i] > arr[i+1]) {
        temp <- arr[i]
        arr[i] <- arr[i+1]
        arr[i+1] <- temp
        swapped <- TRUE
      }
    }
    if (!swapped) {
      break
    }
  }
  return(arr)
}

arr <- c(64, 34, 25, 12, 22, 11, 90)
sorted <- bubble_sort(arr)
print("Sorted array:")
print(sorted)

File: c/sorting_algorithm.clj
(defn bubble-sort [arr]
  (loop [arr arr]
    (let [swapped (atom false)]
      (doseq [i (range (dec (count arr)))]
        (when (> (arr i) (arr (inc i)))
          (swap! swapped true)
          (let [temp (arr i)]
            (aset arr i (arr (inc i)))
            (aset arr (inc i) temp))))
      (if @swapped
        (recur arr)
        arr))))

(def arr [64 34 25 12 22 11 90])
(def sorted (bubble-sort arr))
(println "Sorted array:" sorted)
